zypp/target/rpm/RpmDb.cc

```diff
@@ -1239,6 +1240,44 @@ namespace
 
     if ( ret != RpmDb::CHK_OK )
     {
+      // check was not OK replace the signatures with the long version but reading it from the RPM header
+
+      // Get signature info from the package header, RPM always prints only the 8 byte ID
+      auto header = RpmHeader::readPackage( path_r, RpmHeader::NOVERIFY );
+      if ( !header ) {
+        ERR << "Failed to read package" << std::endl;
+        return RpmDb::CHK_ERROR;
+      }
+
+      auto keyMgr = zypp::KeyManagerCtx::createForOpenPGP();
+      std::unordered_map< std::string, std::string> fprs;
+      const auto &addFprs = [&]( auto tag ){
+        const auto &list1 = keyMgr.readSignatureFingerprints( header->blob_val( tag ) );
+        for ( const auto &id : list1 ) {
+          if ( id.size() <= 8 )
+            continue;
+
+          const std::string keyPrefix( "key ID " );
+          fprs.insert( std::make_pair( keyPrefix+id.substr( id.size() - 8 ), keyPrefix+id ) );
+
+        }
+      };
+
+      addFprs( RPMTAG_SIGGPG );
+      addFprs( RPMTAG_SIGPGP );
+      addFprs( RPMTAG_RSAHEADER );
+      addFprs( RPMTAG_DSAHEADER );
+
+      for ( auto &detail : detail_r ) {
+        auto &line = detail.second;
+
+        // replace the short key ID with the long ones parsed from the header
+        for ( const auto &fpr : fprs ) {
+          str::replaceAll( line, fpr.first, fpr.second );
```

Comment:
As Michael said, rpm may report the full FPR in the future. 
Reading the FPR from the header and replacing the last 8 byte of the reported ID (L1261) must not happen if the reported id is longer than 8.

Reply:
Done